packages/overlay/src/overlay-trigger.ts

```diff
@@ -55,104 +48,44 @@ export class OverlayTrigger extends LitElement {
 
-    public onOverlayOpen(event: Event, interaction: TriggerInteractions): void {
-        const isClick = interaction === 'click';
-        const overlayElement = isClick ? this.clickContent : this.hoverContent;
-
-        /* istanbul ignore if */
-        if (!overlayElement) {
-            return;
-        }
-        if (!overlayRoot) {
-            overlayRoot = new OverlayRoot();
-        }
-
-        const delayAttribute = overlayElement.getAttribute('delay');
-        const delay = delayAttribute ? parseFloat(delayAttribute) : 0;
-
-        const queryThemeDetail: ThemeData = {
-            color: undefined,
-            scale: undefined,
-        };
-        const queryThemeEvent = new CustomEvent<ThemeData>('query-theme', {
-            bubbles: true,
-            composed: true,
-            detail: queryThemeDetail,
-            cancelable: true,
-        });
-        this.dispatchEvent(queryThemeEvent);
-
-        const overlayOpenDetail: OverlayOpenDetail = {
-            content: overlayElement,
-            delay: delay,
-            offset: this.offset,
-            placement: this.placement,
-            trigger: this,
-            interaction: interaction,
-            theme: queryThemeDetail,
-        };
-
-        const overlayOpenEvent = new CustomEvent<OverlayOpenDetail>(
-            'sp-overlay-open',
-            {
-                bubbles: true,
-                composed: true,
-                detail: overlayOpenDetail,
-            }
-        );
-
-        this.dispatchEvent(overlayOpenEvent);
-    }
-
-    public onOverlayClose(
-        event: Event,
-        interaction: TriggerInteractions
-    ): void {
-        const isClick = interaction === 'click';
-        const overlayElement = isClick ? this.clickContent : this.hoverContent;
-
-        /* istanbul ignore if */
-        if (!overlayElement) {
-            return;
-        }
-
-        const overlayCloseDetail: OverlayCloseDetail = {
-            content: overlayElement,
-        };
-
-        const overlayCloseEvent = new CustomEvent<OverlayCloseDetail>(
-            'sp-overlay-close',
-            {
-                bubbles: true,
-                composed: true,
-                detail: overlayCloseDetail,
-            }
-        );
-
-        this.dispatchEvent(overlayCloseEvent);
-    }
-
-    public onTriggerClick(event: Event): void {
+    public onTriggerClick(): void {
         /* istanbul ignore else */
-        if (this.clickContent) {
-            this.onOverlayOpen(event, 'click');
+        if (this.targetContent && this.clickContent) {
+            this.clickOverlay = Overlay.open(
+                this.targetContent,
+                'click',
+                this.clickContent,
+                {
+                    offset: this.offset,
+                    placement: this.placement,
+                }
+            );
         }
     }
 
-    public onTriggerMouseOver(event: Event): void {
+    public onTriggerMouseOver(): void {
```

Comment:
Nit: this method name is out of sync with its listener `@mouseenter=${this.onTriggerMouseOver}` Can we think of a less specific naming here that would be more lenient to ongoing refactoring?

Reply:
I fixed the event name mismatch. I shifted the methods around a bit. I wanted the specific naming because there are three different DOM nodes that we register events on and I wanted it to be obvious which callbacks relate to which nodes